src/reducers/articleGroups.js

```diff
@@ -85,6 +86,15 @@ const articleGroups = (state = {}, action) => {
       });
     }
 
+    case DELETE_ARTICLE_FROM_GROUP: {
+      const { articleId, groupId } = action.payload;
+      const articleIndex = state[groupId].articles.indexOf(articleId);
+
+      state = Object.assign({}, state);
+      state[groupId].articles.splice(articleIndex, 1);
```

Comment:
This is mutating the articles list. Meaning that the previous state is also changed.

Use the `removeFromArticles` helper function.
Also I think it's cleaner to do this in the subreducer I made. Call it like `ADD_ARTICLE_TO_GROUP` does.